### PR TITLE
Drei/views.lisp : space width is zero, use advance-width instead

### LINKED ISSUE #672: method tab-width and cohorts in Libraries/Drei/views.lisp
### the above methods use tab characters without checking for if use-tabs is actually set or not, so tabify gives me mixed files with tab-chars and space-chars, i don't think that's the way it is intended really.
so my solution is below:

```
(defgeneric space-width (stream tabify)
  (:documentation "Return the width of a space character on <stream> in device units (most likely pixels).")
  (:method ((stream extended-output-stream) (tabify tabify-mixin))
           (maybe-update-recordings stream tabify)
           (with-accessors ((recorded-space-width recorded-space-width)) tabify
                           recorded-space-width)))
(defgeneric tab-width (stream tabify)
  (:documentation "Return the width of a tab character on <stream> in device units (most likely pixels).")
  (:method ((stream extended-output-stream) (tabify tabify-mixin))
           (with-accessors ((tab-space-count tab-space-count) 
                                               (recorded-tab-width recorded-tab-width) (use-tabs use-tabs)) tabify
                           (if use-tabs
                               recorded-tab-width
                               (* (if tab-space-count tab-space-count 8) (space-width stream tabify))))))

(defgeneric next-tab-stop (stream tabify x)
  (:documentation "Return the distance to the next tab-stop after 'x' on <stream> in device units (most likely pixels).")
  (:method ((stream extended-output-stream) (tabify tabify-mixin) x)
           (with-accessors ((tab-space-count tab-space-count) 
                                               (tab-stops tab-stops) (use-tabs use-tabs)) tabify
                           (flet ((round-up (x width)
                                            (if (= 0 width)
                                                (* tab-space-count (space-width stream tabify))
                                              (- width (mod x width)))))
                                 (if use-tabs
                                     (when tab-stops
                                       (let ((next (find-if (lambda (pos) (> pos x)) tab-stops)))
                                         (or (and next (- next x)) (round-up x (tab-width stream tabify)))))
                                   (round-up x
                                                        (* (if tab-space-count tab-space-count 8) (space-width stream tabify))))))))

(defgeneric (setf tab-stop-columns) (column-list tabify)
  (:documentation "Set the TAB-STOPS of view at the character column offsets in <column-list>.")
  (:method (column-list (tabify tabify-mixin))
           (with-accessors ((tab-stops tab-stops) (recorded-stream recorded-stream)) tabify
    (setf tab-stops
          (and column-list
               (sort (mapcar (lambda (col) (* col (space-width recorded-stream tabify)))
                             column-list)
                     #'<))))))
```

### LINKED ISSUE #678: tabify-buffer-region in Libraries/Drei/base.lisp is slightly wrong
I tried M-x Tabify Region many times over in Climacs, and some words looked almost like written together.
But upon closer inspection one could observe the Cursor disappear in between the words.
I thought maybe it's a font-issue, or some issue with presentation types. I looked at the file via emacs in a terminal and there was no hint of something being wrong. The almost together written words had a real space in there. 
The Bug also made the Climacs window split upon completion.

It's some off by one error. No resolution yet!

### PR DESCRIPTION
**Pull request by nsrahmad**
Fixes #678 and fixes #672

For context see above issues.
To try this in Climacs:
1) `(ql:quickload 'climacs)`.
2) `(climacs:climacs)`
3) Open a text file with `C-x C-f`. Press `C-x h` to select all region.
4) run `M-x tabify-region`, and save with `C-x C-f`.
You might want to inspect file in Emacs with `M-x whitespace-mode`. with this PR `tabify-region` and `untabify-region` should work without errors in Climacs.

### CODE REVIEW

File: Libraries/Drei/views.lisp [hunk 1]
```diff
@@ -73,7 +73,10 @@ If empty, tabs every TAB-WIDTH are assumed.")))
                    (recorded-stream recorded-stream)) tabify
     (unless (eq stream recorded-stream)
       ;; Update the recorded values.
-      (setf space-width (stream-character-width stream #\Space)
```
Comment (dkochmanski) [inline Libraries/Drei/views.lisp]: please use `text-style-character-width` for that (also in climb package).

Comment (nsrahmad) [reply Libraries/Drei/views.lisp]: done

File: Libraries/Drei/views.lisp [hunk 2]
```diff
@@ -73,7 +73,9 @@ If empty, tabs every TAB-WIDTH are assumed.")))
...
+      (setf space-width (climb:text-style-character-width (medium-text-style stream)
+							  stream
+							  #\Space)
```
Comment (dkochmanski) [inline Libraries/Drei/views.lisp]: please detabify the file (changes contain tabs). I've missed that part in the title-pane PR - you may want to tweak your editor settings. I believe that this is enough for emacs:
`(setq-default indent-tabs-mode nil)`

Comment (nsrahmad) [reply Libraries/Drei/views.lisp]: done.

**Comment by dkochmanski** [comment]
merged, thanks!

**Pull request merged**